lib/resque_message_handler.rb

```diff
@@ -125,19 +125,24 @@ def update
   private
 
   # Given a hash relating barcodes to scsb items, returns a new hash consisting
-  # of only those items that have the named status (`:available` or
-  # `:unavailable`). To ensure a clean segmentation, we only match `:available`
-  # to `"availality": "Available"`
-  def select_by_status(barcode_mapping, status)
+  # of only those items that are either:
+  #  1. Available
+  #  2. Incomplete
+  # These are the classes of records we're able to update.
+  def get_barcodes_allowing_updates(barcode_mapping)
     barcode_mapping.select do |barcode, item|
-      if status == :available
-        item['availability'] == 'Available'
-      else
-        item['availability'] != 'Available'
-      end
+      item['availability'] == 'Available' || item['title'] == 'Dummy Title'
     end
   end
 
+  # Get inverse of get_barcodes_allowing_updates
+  def get_barcodes_disallowing_updates(barcode_mapping)
+    allowed = get_barcodes_allowing_updates barcode_mapping
+    # Build hash from difference obtained by deleting everything from
+    # `barcode_mapping` that exists in `allowed`:
+    Hash[*(barcode_mapping.to_a - allowed.to_a).flatten]
+  end
```

Comment:
I would prefer to use a reject here. I.e. get_barcodes_allowing updates = array.select{|blah| blah(blah)}, get_barcodes_disallowing_updates = array.reject{|blah| blah(blah)}, where blah could be a separate method